app/src/main/java/com/pajato/android/gamechat/exp/Checkerboard.java

```diff
@@ -73,6 +73,16 @@
 
     /** Build an instance to establish the cell size for a given context. */
     Checkerboard(final Context context) {
+        // Establish the controls height. Remember to convert the SP of text in the layout to DP.
```

Comment:
Ok, it is clear what you are doing with the controls height.  You want to remove the class variable (in general, programs are better that have fewer variables, preferring method variables over instance or class variables).  In this case you might want to move the calculation of the controls height to a private instance method (passing in the metrics object) that is called in getCellSize() exactly when it is needed.